sdk/python/tests/integration/feature_repos/test_repo_configuration.py

```diff
@@ -52,8 +57,108 @@ class TestRepoConfig:
 PROVIDERS: List[str] = []
 
 
+@dataclass
+class Environment:
+    name: str
+    test_repo_config: TestRepoConfig
+    feature_store: FeatureStore
+    data_source: DataSource
+    data_source_creator: DataSourceCreator
+
+    end_date = datetime.now().replace(microsecond=0, second=0, minute=0)
+    start_date = end_date - timedelta(days=7)
+    before_start_date = end_date - timedelta(days=365)
+    after_end_date = end_date + timedelta(days=365)
+
+    customer_entities = list(range(1001, 1110))
+    customer_df = driver_test_data.create_customer_daily_profile_df(
+        customer_entities, start_date, end_date
+    )
+
+    driver_entities = list(range(5001, 5110))
+    driver_df = driver_test_data.create_driver_hourly_stats_df(
+        driver_entities, start_date, end_date
+    )
+
+    orders_df = driver_test_data.create_orders_df(
+        customers=customer_entities,
+        drivers=driver_entities,
+        start_date=before_start_date,
+        end_date=after_end_date,
+        order_count=1000,
+    )
+
+    def customer_fixtures(self) -> FeatureView:
+        customer_table_id = self.data_source_creator.get_prefixed_table_name(
+            self.name, "customer_profile"
+        )
+        ds = self.data_source_creator.create_data_sources(
+            customer_table_id,
+            self.customer_df,
+            event_timestamp_column="event_timestamp",
+            created_timestamp_column="created",
+        )
+        return create_customer_daily_profile_feature_view(ds)
+
+    def driver_stats_fixtures(self) -> FeatureView:
+        driver_table_id = self.data_source_creator.get_prefixed_table_name(
+            self.name, "driver_hourly"
+        )
+        ds = self.data_source_creator.create_data_sources(
+            driver_table_id,
+            self.driver_df,
+            event_timestamp_column="event_timestamp",
+            created_timestamp_column="created",
+        )
+        return create_driver_hourly_stats_feature_view(ds)
+
+    def order_fixtures(self) -> pd.DataFrame:
+        return self.orders_df
+
+    def orders_sql_fixtures(self) -> Optional[str]:
+        pass
+
+
+def vary_full_feature_names(configs: List[TestRepoConfig]) -> List[TestRepoConfig]:
+    new_configs = []
+    for c in configs:
+        true_c = replace(c, full_feature_names=True)
+        false_c = replace(c, full_feature_names=False)
+        new_configs.extend([true_c, false_c])
```

Comment:
So, the configs argument *has* the params  but they are  set to a default value so that the config is still valid for a test to use. 

Also just to note this is only adding configs to the `new_configs` list which doesn't get the original value, so I was hoping that i could get away with it without having to worry about *what* the default value was exactly.

Bug, instead of doing this replace twice, would you prefer that I just do a `replace` with the non-default value? That will probably read better.

Reply:
I actually ended up leaving this as is - I think it's better to be comprehensive in the vary methods, and I think it reads okay - Lemme know if you feel strongly against